sympy/solvers/ode/tests/test_ode.py

```diff
@@ -2883,6 +2883,43 @@ def test_separable_reduced():
     assert sol == Eq(log(x**2*f(x))/2 - log(x**2*f(x) - 2)/2, C1 + log(x))
     assert checkodesol(eq, sol, order=1, solve_for_func=False)[0]
 
+    eq = Eq(f(x).diff(x) + f(x)/x * (1 + (x**(S(2)/3)*f(x))**2), 0)
+    # FIXME: This one hangs
+    #sol = dsolve(eq, hint = 'seprable_reduced')
+    # assert checkodesol(eq, sol) = (True, 0)
+
+    eq = Eq(f(x).diff(x) + f(x)/x * (1 + (x*f(x))**2), 0)
+    sol = dsolve(eq, hint = 'separable_reduced')
+    assert sol == [Eq(f(x), -sqrt(2)*sqrt(1/(C1 + log(x)))/(2*x)),\
+                   Eq(f(x), sqrt(2)*sqrt(1/(C1 + log(x)))/(2*x))]
+    assert checkodesol(eq, sol) == [(True, 0)]*2
+
+    eq = Eq(f(x).diff(x) + (x**4*f(x)**2 + x**2*f(x))*f(x)/(x*(x**6*f(x)**3 + x**4*f(x)**2)), 0)
+    sol = dsolve(eq, hint = 'separable_reduced')
+    assert sol == Eq(f(x), C1 + 1/(2*x**2))
+    assert checkodesol(eq, sol) == (True, 0)
+
+    eq = Eq(f(x).diff(x) + (f(x)**2)*f(x)/(x), 0)
+    sol = dsolve(eq, hint = 'separable_reduced')
+    assert sol == [Eq(f(x), -sqrt(2)*sqrt(1/(C1 + log(x)))/2),\
+                  Eq(f(x), sqrt(2)*sqrt(1/(C1 + log(x)))/2)]
+
+    eq = Eq(f(x).diff(x) + (f(x)+3)*f(x)/(x*(f(x)+2)), 0)
+    # FIXME: This one hangs
+    #sol == dsolve(eq, hint = 'separable_reduced')
+    #assert checkodesol(eq, sol) == (True, 0)
```

Comment:
It would be good to add the link in a comment (if this is the same issue):
```
https://github.com/sympy/sympy/issues/19124
```
Also we can still put what the solution should be (if we know what it should be). We can get a solution here with `simplify=False` so that should be tested as well:
```julia
In [63]: eq = Eq(f(x).diff(x) + (f(x)+3)*f(x)/(x*(f(x)+2)), 0)                                                                    

In [64]: sol = dsolve(eq, simplify=False)                                                                                         

In [65]: sol                                                                                                                      
Out[65]: 
log(f(x) + 3)   2⋅log(f(x))              
───────────── + ─────────── = C₁ - log(x)
      3              3                   

In [66]: checkodesol(eq, sol, solve_for_func=False)                                                                               
Out[66]: (True, 0)
```